### PR TITLE
Add documentation for two AMD APUs

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: docs/source/keywords.rst [hunk 1]
```diff
@@ -599,6 +604,11 @@ Kokkos will attempt to autodetect the architecture flag at configuration time.
       * GFX906
       * MI50, MI60
       * (since Kokkos 4.2)
+
+    * * ``Kokkos_ARCH_AMD_GFX1103``
+      * GFX1103
+      * Phoenix APU
```
Comment (dalg24) [inline docs/source/keywords.rst:610]: As I understand it, "Phoenix" is the codename for the microarchitecture whereas this column says "Models".
Should we list a couple popular models, something like "Phoenix APUs (e.g. MODEL-1, MODEL-2, etc.)"

File: docs/source/keywords.rst [hunk 2]
```diff
@@ -575,10 +575,15 @@ Kokkos will attempt to autodetect the architecture flag at configuration time.
...
+    * * ``Kokkos_ARCH_AMD_GFX942_APU``
+      * GFX942
+      * MI300A
+      * (since Kokkos 4.5)
+
     * * ``Kokkos_ARCH_AMD_GFX942``
       * GFX942
       * MI300A, MI300X
-      * (since Kokkos 4.2)
+      * (since Kokkos 4.2, since Kokkos 4.5 this should only be used for MI300X)
```
Comment (dalg24) [inline docs/source/keywords.rst:586]: I haven't looked at how this renders but I expects this folds over multiple lines.
Did it still look fine to you, should we instead do a "see below"?

Comment (Rombur) [reply docs/source/keywords.rst:586]: Looks ok to me:
image

Comment (dalg24) [reply docs/source/keywords.rst:586]: Sure

File: docs/source/keywords.rst (hunk 1 above)
Comment (Rombur) [reply docs/source/keywords.rst]: I have added something

**Review by masterleinad — APPROVED** [review]
Fine with me.

**Pull request merged**